include/ringmesh/basic/frame.h

```diff
@@ -147,11 +147,79 @@ namespace RINGMesh
             return origin_;
         }
 
+        vecn< DIMENSION > coords_to_local(
+            const vecn< DIMENSION >& global_coords ) const;
+
+        vecn< DIMENSION > coords_to_global(
+            const vecn< DIMENSION >& local_coords ) const
+        {
+            vecn< DIMENSION > global_coords = this->origin();
+            for( auto coord : range( DIMENSION ) )
+            {
+                for( auto coor : range( DIMENSION ) )
+                {
+                    global_coords[coord] +=
+                        local_coords[coor] * this->operator[]( coor )[coord];
+                }
+            }
+            return global_coords;
+        }
+
     private:
         vecn< DIMENSION > origin_{};
     };
     ALIAS_2D_AND_3D( ReferenceFrame );
 
+    class RINGMESH_API CartesianGridFrame3D : public ReferenceFrame< 3 >
```

Comment:
Is this name generic enough? This means the class is related to cartesian grid but it is just a special ReferenceFrame used in a cartesian grid. What is so special about this frame?

Reply:
We use it to store a ReferenceFrame3D to be able to calculate more easily the coordinate change when passing from global to local (so as not to have to invert a 3x3 matrix for every point we import for exemple). We could also call it InvertibleFrame3D or something like that ?

Reply:
I think you should find another name but I am not sure which name...

Reply:
EasyInvertibleFrame3D ?